app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -56,6 +56,11 @@ def refresh
         }
         scope :with_content, -> { with_features(PULP_FEATURE, PULP_NODE_FEATURE) }
 
+        scope :with_repo, lambda do |repo|
```

Comment:
So my rails server wasn't starting with this scope definition:

```
/home/vagrant/katello/app/models/katello/concerns/smart_proxy_extensions.rb:59:in `lambda': tried to create Proc object without a block (ArgumentError)
```

I looked at the docs and read "Using a class method is the preferred way to accept arguments for scopes." (https://guides.rubyonrails.org/active_record_querying.html#passing-in-arguments) and I think using a class method would be consistent with other areas we are doing similar things